### PR TITLE
feat(gp): port to gaussx 0.0.14 — delegate Kalman/RTS, drop autocov shim

### LINKED ISSUE #124: refactor: thin pyrox to consume promoted gaussx primitives
## Context

A reconciliation pass identified pyrox primitives that should be promoted to gaussx (tracked in jejjohnson/gaussx#130). Once those land, this issue tracks the pyrox-side cleanup: removing duplicates and switching to gaussx imports.

Will be broken into smaller PRs once the gaussx side is in place.

---

## 1. Replace `_stable_cholesky` with `gaussx.safe_cholesky` (cheap win, can ship now)

gaussx already exports `safe_cholesky`. Pyrox's `_stable_cholesky` is a parallel implementation — adaptive jitter Cholesky on raw arrays.

**Files to update:**
- [src/pyrox/gp/_inference_nongauss.py](https://github.com/jejjohnson/pyrox/blob/main/src/pyrox/gp/_inference_nongauss.py) — definition at L239, callsites at L112, L202, L314, L319, L763
- Grep for any other callers before removing

This one does **not** depend on the gaussx promotion issue and can land immediately.

## 2. Remove `GaussHermite` wrapper (or document its value)

[src/pyrox/gp/_integrators.py#L33-L65](https://github.com/jejjohnson/pyrox/blob/main/src/pyrox/gp/_integrators.py#L33) is a thin wrapper around `gauss_hermite_points` + vmap. gaussx exports `GaussHermiteIntegrator` directly.

Decide: delete the wrapper and re-export `gaussx.GaussHermiteIntegrator`, or keep it with a docstring explaining why (e.g., NumPyro-specific contract).

## 3. Re-export SDE kernels from gaussx (depends on gaussx#TBD item 1)

Once `gaussx._kernels.sde` exists, replace [src/pyrox/gp/_sde_kernels.py](https://github.com/jejjohnson/pyrox/blob/main/src/pyrox/gp/_sde_kernels.py) with a re-export shim, or remove entirely and update [src/pyrox/gp/__init__.py](https://github.com/jejjohnson/pyrox/blob/main/src/pyrox/gp/__init__.py) to pull from gaussx.

Same treatment for the `SDEKernel` protocol in [src/pyrox/gp/_protocols.py](https://github.com/jejjohnson/pyrox/blob/main/src/pyrox/gp/_protocols.py).

## 4. Use gaussx Kalman/RTS core (depends on gaussx#TBD item 2)

[src/pyrox/gp/_markov.py](https://github.com/jejjohnson/pyrox/blob/main/src/pyrox/gp/_markov.py) `_kalman_filter` and `_rts_smoother` should delegate to `gaussx._ssm`. Keep pyrox's masking trick as a thin wrapper on top.

## 5. Use `gaussx._sde_autocov_pairs` (depends on gaussx#TBD item 3)

Drop [src/pyrox/gp/_sparse_markov.py#L60-L81](https://github.com/jejjohnson/pyrox/blob/main/src/pyrox/gp/_sparse_markov.py#L60-L81) once promoted.

## 6. Use `gaussx` EP tilted moments (depends on gaussx#TBD item 4)

Drop [src/pyrox/gp/_inference_nongauss.py#L273-L289](https://github.com/jejjohnson/pyrox/blob/main/src/pyrox/gp/_inference_nongauss.py#L273) `_ep_tilted_moments` and import from gaussx.

## 7. Thin likelihood `log_prob`s to gaussx delegates (depends on gaussx#TBD item 5)

Once gaussx has bare exp-family `log_prob` math, [src/pyrox/gp/_likelihoods.py](https://github.com/jejjohnson/pyrox/blob/main/src/pyrox/gp/_likelihoods.py) keeps the parameterised / prior'd wrappers but delegates the density math.

## 8. Replace open-coded per-site natural-param conversions (depends on gaussx#TBD item 6)

Throughout [src/pyrox/gp/_inference_nongauss.py](https://github.com/jejjohnson/pyrox/blob/main/src/pyrox/gp/_inference_nongauss.py) and [src/pyrox/gp/_inference_nongauss_markov.py](https://github.com/jejjohnson/pyrox/blob/main/src/pyrox/gp/_inference_nongauss_markov.py), the `(λ⁽¹⁾, Λ⁽²⁾) = (J − Hm, −H)` arithmetic should route through gaussx scalar/diagonal helpers.

---

## Verification gates (each PR must pass)

- `uv run pytest -v -m "not slow"` (fast suite)
- `uv run --group lint ruff check .` and `ruff format --check .`
- `uv run --group typecheck ty check src/pyrox`
- Spot-check the Markov GP demo notebooks still execute end-to-end after Kalman/SDE migrations.

## Order of operations

1. Item 1 (cheap, independent) — can land first.
2. Item 2 (independent) — quick decision.
3. Items 3–8 — gated on the corresponding gaussx promotion PRs.

### PR DESCRIPTION
**Pull request by jejjohnson**
## Summary

Bumps gaussx from a `main`-tracking direct dep to a pinned `v0.0.14` tag and ports the pyrox-side Markov-GP internals to consume the upstream APIs released alongside this version. Closes [pyrox#124](https://github.com/jejjohnson/pyrox/issues/124) items 4 and 5; items 1–3 and 6–8 already landed in prior PRs (`safe_cholesky`, `GaussHermiteIntegrator`, SDE-kernel re-exports, `ep_tilted_moments`, `damped_natural_update`).

## Changes

- **`pyproject.toml`** — pyrox `0.0.12 → 0.0.13`; gaussx pin moves from `branch = "main"` to `tag = "v0.0.14"` and the PEP 508 dep bumps to `gaussx @ git+...@v0.0.14`.

- **`src/pyrox/gp/_markov.py`** — `_kalman_filter` and `_rts_smoother` are now thin shims over `gaussx.kalman_filter` / `gaussx.rts_smoother`. gaussx 0.0.13 added the time-varying / per-step `mask` generalisation that pyrox needs, and 0.0.14 fixed the parallel variant; pyrox no longer carries its own forward/backward implementations. Function signatures are kept so `_inference_nongauss_markov` and the existing direct-import test (`test_filter_stays_finite_when_masked_step_has_degenerate_S`) keep working — the zero-`R` / zero-`Q` corner case is handled by gaussx's `lax.cond`-gated update, which never enters the Cholesky path on masked steps.

- **`src/pyrox/gp/_sparse_markov.py`** — drops the local `_sde_autocov_pairs` helper (a 2-line wrapper over `gaussx.sde_autocovariance`) and inlines the upstream call at both call sites (`inducing_operator`, `cross_covariance`).

Net: ~150 lines of pyrox-side Kalman / autocov plumbing deleted; behaviour unchanged.

## Test plan

Run locally against a fresh `uv sync` of the v0.0.14 pin:

- [x] `uv run pytest -m 'not slow' --no-cov --ignore=docs` — **764 passed, 51 deselected**
- [x] `uv run pytest tests/gp/test_markov_gp.py tests/gp/test_sparse_markov.py -m 'not slow' --no-cov` — 20 passed
- [x] `uv run --group lint ruff check .` — clean
- [x] `uv run --group lint ruff format --check .` — 121 files already formatted
- [x] `uv run --group typecheck ty check src/pyrox` — clean

## Note on `uv.lock`

The lockfile update (gaussx 0.0.11 → 0.0.14, pyrox 0.0.12 → 0.0.13, ~30 effective lines) couldn't ride along in this push due to a payload-size limit on the API client used to commit. The CI workflow uses `uv sync --group dev --extra optax --extra bnf` without `--frozen`, so it resolves from `pyproject.toml` directly and regenerates the lockfile in-place — but please run `uv lock` locally before merge to commit a deterministic snapshot.

https://claude.ai/code/session_013bc5XZr5u5MXxTyTn1Amyp

---
_Generated by [Claude Code](https://claude.ai/code/session_013bc5XZr5u5MXxTyTn1Amyp)_

### CODE REVIEW

File: src/pyrox/gp/_markov.py [hunk 1]
```diff
@@ -27,9 +27,11 @@
 from typing import TYPE_CHECKING, Protocol
 
 import equinox as eqx
+import gaussx
 import jax
 import jax.numpy as jnp
 import numpyro
+from gaussx._ssm._kalman import FilterState
```
[the remark this answers is not in the document]
Comment (jejjohnson) [reply src/pyrox/gp/_markov.py]: Addressed in 6e929ff. Switched to `from gaussx import FilterState` (the public re-export landed in gaussx 0.0.13 / is in 0.0.14). The original ty-inference issue that nudged me to the private path is independently fixed by the explicit `filter_state: FilterState = FilterState(...)` annotation, so the public import now works cleanly without further suppressions. Verified locally: `ty check src/pyrox` clean, `ruff check` clean, targeted Markov-GP tests pass.

---
_Generated by [Claude Code](https://claude.ai/code)_
